comp/core/workloadmeta/collectors/internal/kubelet/kubelet.go

```diff
@@ -436,6 +437,19 @@ func extractEnvFromSpec(envSpec []kubelet.EnvVar) map[string]string {
 	return env
 }
 
+// filterEnvVars removes unsupported env var sources (eg. ConfigMap, Secrets, etc.)
+func filterEnvVars(envSpec *[]kubelet.EnvVar) {
```

Comment:
It should be possible to rewrite this in a cleaner and more standard way using the `slices` stdlib package.
You can check `DeleteFunc`: https://pkg.go.dev/slices#DeleteFunc